autoextract/aio/retry.py

```diff
@@ -95,13 +105,22 @@ def __call__(self, retry_state: RetryCallState) -> float:
             return self.network_wait(retry_state=retry_state)
         elif _is_server_error(exc):
             return self.server_wait(retry_state=retry_state)
+        elif _is_retriable_query_error(exc):
+            try:
+                wait = exc.retry_seconds
+            except ValueError as exc:
+                logger.warning(exc)
+                return 300  # 5 minutes
+
+            return wait or self.retriable_wait(retry_state=retry_state)
         else:
             raise RuntimeError("Invalid retry state exception: %s" % exc)
 
 
 class autoextract_stop_strategy(stop_base):
     def __init__(self):
         self.stop_on_throttling_error = stop_never
+        self.stop_on_retriable_query_error = stop_never
```

Comment:
This is something I wanted to discuss with you guys. Even with the domain occupied errors, I guess it's somewhat dangerous to ask for perpetual retries. Even the 15 minutes default stop time sounds like too much to me. Should we be waiting that long for responses? Another option is setting a max number of retries (e.g. 3 retries).

Reply:
> Should we be waiting that long for responses?

I think it depends. Let's imagine that you receive a 500. The site could be down, and it could take a while until it comes back to life. 

In any case, probably this should be configurable somehow. Some user might need faster responses so setting a lower timeout should be possible. But I guess this is out of the scope of this ticket. 

> Another option is setting a max number of retries

I prefer using timing and jitter. It would be beneficial for the API.

Reply:
With last commit I make sure we're following your original implementation:

- retry domain occupied errors forever
- retry other query-level errors up to 15 minutes